test/compojure/api/integration_test.clj

```diff
@@ -71,8 +78,11 @@
   "If request has query-param x, presume it's a integer and multiply it by two
    before passing request to next handler."
   [handler]
-  (fn [req]
-    (handler (update-in req [:query-params "x"] #(* (Integer. %) 2)))))
+  (fn
```

Comment:
Asyncifying middlewares seems to tricky, because you have to call the inner handler in the correct fashion and I don't see how to do it automatically.